spec/observables/dom/ajax-spec.ts

```diff
@@ -1124,10 +1032,365 @@ describe('ajax', () => {
       });
     });
   });
+
+  describe('with includeDownloadProgress', () => {
+    it('should emit download progress', () => {
+      const results: any[] = [];
+
+      ajax({
+        method: 'GET',
+        url: '/flibbertyJibbett',
+        includeDownloadProgress: true,
+      }).subscribe({
+        next: (value) => results.push(value),
+        complete: () => results.push('done'),
+      });
+
+      const mockXHR = MockXMLHttpRequest.mostRecent;
+      mockXHR.respondWith(
+        {
+          status: 200,
+          total: 5,
+          loaded: 5,
+          responseType: 'json',
+          responseText: JSON.stringify({ boo: 'I am a ghost' }),
+        },
+        { uploadProgressTimes: 5, downloadProgressTimes: 5 }
+      );
+
+      const request = {
+        async: true,
+        body: undefined,
+        crossDomain: true,
+        headers: {
+          'x-requested-with': 'XMLHttpRequest',
+        },
+        includeDownloadProgress: true,
+        method: 'GET',
+        responseType: '',
+        timeout: 0,
+        url: '/flibbertyJibbett',
+        withCredentials: false,
+      };
+
+      expect(results).to.deep.equal([
+        {
+          type: 'download_loadstart',
```

Comment:
I thought about it. I think that we have to do `enum` and not `const enum`, and therefore we'd end up exporting an object. That doesn't really mesh with how events work elsewhere. I'll consider this a vote in favor of an enum, though.

Reply:
FWIW, my vote would be against exporting an enum. If something is to be exported, I'd vote for a type that's a union of literals, rather than an enum.